(RQR12/24) Fix license case and explain GUIDs in comment

SonicFrog opened the pull request:
It seems we do not need need additionnal GUIDs since it is possible to split the signed/unsigned variant in two separate components (each with its own constraints on bootloader version). This will require a small patch in fwupd though.

Signed-off-by: Ogier Bouvier <obouvier@logitech.com>

hughsie commented on MPK01/MPK01.02/lvfs/com.logitech.MPK01.metainfo.xml:
```diff
@@ -15,7 +15,7 @@
   <url type="homepage">http://www.logitech.com/en-us/product/k780-multi-device-wireless-keyboard</url>
   <metadata_license>CC0-1.0</metadata_license>
-  <project_license>Proprietary</project_license>
+  <project_license>proprietary</project_license>
```
I guess this is right, but you might as well use the SPDX `LicenseRef-proprietary` instead if you want to me super clear.

hughsie commented on RQR24/RQR24.03/lvfs/com.logitech.Unifying.RQR24.metainfo.xml:
```diff
@@ -12,6 +12,8 @@
     </p>
   </description>
+  <!-- Signed/unsigned versions are split into different components to permit
+       having different <requires> on each even though they share GUIDs -->
```
Makes sense to me.

Commit:
RQR12/24, fix license case and explain GUIDs in comment

Using LicenseRef-Proprietary instead of proprietary on Richard's
advice.

Also adding a comment to explain how the GUIDs are handled

Signed-off-by: Ogier Bouvier <obouvier@logitech.com>

ogay commented:
I'm squashing this commit with the last commit of PR#8 ("fix license case" 48b8126).
